src/components/Dropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  FC,
+  useState,
+  ChangeEvent,
+  FocusEvent,
+  memo,
+  useCallback,
+} from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { Person } from '../types/Person';
+
+import './Dropdown.scss';
+
+interface Props {
+  people: Person[];
+  delay?: number;
+  onChangeQuery: (query: string) => void;
+  onSelectPerson: (person: Person | null) => void;
+}
+
+const Dropdown: FC<Props> = ({
+  people,
+  delay = 300,
+  onSelectPerson,
+  onChangeQuery,
+}) => {
+  const [openDropdown, setOpenDropdown] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+
+  const debouncedOnChangeQuery = useCallback(
+    debounce((str: string) => {
+      onChangeQuery(str);
+    }, delay),
+    [delay, onChangeQuery],
+  );
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(event.target.value);
+    onSelectPerson(null);
+    debouncedOnChangeQuery(event.target.value);
+  };
```

Comment:
Follow naming conventions for methods. The method `handleQueryChange` should be named using the `on` prefix, such as `onQueryChange`, to align with common React event handler conventions.